### ISSUE TITLE
Incorporate a loading animation when pasting in large datasets

### ISSUE THREAD

**Issue by adidottxt** [jspreadsheet/ce#794]
Hi there,

I'm currently working on a project with Jexcel where I have an `onafterchanges` handler that runs each time a change is made to the JExcel table, which in turn calls another function that saves each changed or row to a local state. This works well for smaller updates to the table, but essentially leaves the page hanging for ~30 seconds if I, say, add 500 rows to the table.

I was wondering if there's a way to incorporate a loading animation for these specific cases; I've tried using onbeforepaste to set a boolean to render a [Loader](https://react.semantic-ui.com/elements/loader/), but unfortunately because the page hangs while the handler function runs, React doesn't seem to re-render the page to switch to the Loader. I've also set `lazyloading` to `True`, but I'd imagine that only works post reading in the data.

Thanks!

**Comment by pphod** [comment]
You can add:
onbeforechange: function() {
jSuites.loading.show();
}
onafterchanges:function() {
jSuites.loading.hide();
}

**Comment by adidottxt** [comment]
@paulhodel Thank you for your response -- while I've got the jSuites loading animation to work, even with setting it up within the `onbeforechange` or `onbeforepaste` handler only results in the loading animation to render after the paste is complete.

I'm wondering if this might be down to how I've currently got the Jexcel table initialized in React (within `useEffect()` because initializing a state update on an unmounted component results in an error). I've also tried `useLayoutEffect()` which is meant to run synchronously immediately after React has performed all DOM mutations, but that doesn't seem to work either. The weird bit is that the actual event handlers seem to execute at the same time (checking via simple console.log statements), but it seems like the actual render remains asynchronous no matter what.
